bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleEngineImpl.java

```diff
@@ -900,13 +895,13 @@ private void setStatus(String ruleUID, RuleStatusInfo newStatusInfo) {
     protected void scheduleRuleInitialization(final String rUID) {
         Future<?> f = scheduleTasks.get(rUID);
         if (f == null || f.isDone()) {
-            scheduleTasks.put(rUID, getScheduledExecutor().schedule(() -> {
+            scheduleTasks.put(rUID, getScheduledExecutor().submit(() -> {
                 final WrappedRule managedRule = getManagedRule(rUID);
                 if (managedRule == null) {
                     return;
                 }
                 setRule(managedRule);
-            }, scheduleReinitializationDelay, TimeUnit.MILLISECONDS));
+            }, TimeUnit.MILLISECONDS));
```

Comment:
This seems like a `schedule` parameter left over:

```suggestion
            }));
```